### PR TITLE
Truncation warning

### PR DESCRIPTION
**Pull request by bkentel**
Fixed a warning about truncating an `int` to a `char`, however the code is problematic in that it won't play well with localization. What should be done here as a better fix?
- moved magic numbers out to named constexprs.
- wrapped long lines, and removed redundant brackets, added const.
- gave two long repeated expressions named const temporaries.

### CODE REVIEW

File: src/iexamine.cpp [hunk 1]
```diff
@@ -2005,33 +2005,41 @@ void iexamine::pick_plant(player *p, map *m, int examx, int examy,
...
     }
+    
     //if the fruit is not ripe yet
-    if (calendar::turn.get_season() != m->get_ter_harvest_season(examx, examy)) {
+    auto const harvest_season = m->get_ter_harvest_season(examx, examy);
+    if (current_season != harvest_season) {
         std::string fruit = item::nname(m->get_ter_harvestable(examx, examy), 10);
-        fruit[0] = toupper(fruit[0]);
-        add_msg(m_info, _("%s ripen in %s."), fruit.c_str(), season_name[m->get_ter_harvest_season(examx, examy)].c_str());
+        fruit[0] = static_cast<char>(toupper(fruit[0])); // TODO: this is not localization friendly
```
Comment (narc0tiq) [inline src/iexamine.cpp]: Is that actually ever displayed anywhere, though? No need to localize a purely internal string.

**Comment by narc0tiq** [comment]
I seem to recall we had this localization issue before, as well, and we didn't find a good answer for that one, either. For now we can just let it stand, I guess.

**Comment by bkentel** [comment]
Nevermind the trying to capitalize one byte of a potentially multi-byte codepoint, this is just wrong for languages with a sentence order different from English... almost seems like one would need a special version of printf that understands extra formatting.

**Comment by KA101** [comment]
OK, on second read 'tis better to leave in the warning than to break localization.

**Comment by bkentel** [comment]
@KA101 the code is broken w.r.t localization with or without this change; this just gets rid of a warning and cleans up the code a tad.

**Comment by kevingranade** [comment]
You're silencing a warning about something that is actually a problem,
that's not a plus.  The simple fix is to use a unicode-aware toupper
method, which I'm sure exists.

**Comment by bkentel** [comment]
There's a locale aware version, but I'm not sure that's what ought to used here. If you can point me in the direction of what should be used here, I'll make the change, otherwise feel free to close this.

**Comment by KA101** [comment]
The PR hopper is horribly backlogged, so closing pending a better solution.  If/when we have one, feel free to reopen.

**Pull request closed without merging**